### ISSUE TITLE
Rule proposal: `export-function-style`

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#154]
Idea from: https://github.com/sindresorhus/refined-github/pull/1077#discussion_r166941975

You can currently export a function using the following with ES modules:

```js
export default function () { … }

export function foo() { … }
```

or

```js
export default () => { … }

export const foo = () => { … }
```

I think the rule should let you configure default and named exports individually, as someone might want to do:

```js
export default () => { … }

export function foo() { … }
```

Thoughts? And does it make sense to configure them individually, like suggested above?

---

// @bfred-it FYI

**Comment by golopot** [comment]
Should this rule ignore functions using `this`?

**Comment by bfred-it** [comment]
I think this rule would be a copy of the existing `function-style`, except localized to exports, so `this` logic is in there.

**Comment by golopot** [comment]
There is a third case `export const foo = function foo() {}`, is it ok to always disallow it?

**Comment by bfred-it** [comment]
These two are equivalents
```js
export const foo = function foo() {};
export const foo = () => {};
```

Both have `foo.name === 'foo'`

**Comment by fregante** [comment]
Default exports are covered by `import/no-anonymous-default-export`, meaning that the only accepted one-line form is:

```js
export default function yourFuctionWithAName () {}
```

Named exports should just be covered by https://github.com/sindresorhus/eslint-plugin-unicorn/issues/3; I don't think exports deserve their own style definition.

Reopen if you disagree.
